lib/NIF/NifFile.cpp

```diff
@@ -976,159 +976,163 @@ OptResult NifFile::OptimizeFor(OptOptions& options) {
 					result.shapesVColorsRemoved.push_back(shapeName);
 
 				BSTriShape* bsOptShape = nullptr;
-				auto bsSegmentShape = dynamic_cast<BSSegmentedTriShape*>(shape);
-				if (bsSegmentShape) {
-					bsOptShape = new BSSubIndexTriShape();
-				}
-				else {
-					if (options.headParts)
-						bsOptShape = new BSDynamicTriShape();
-					else
-						bsOptShape = new BSTriShape();
-				}
-
-				bsOptShape->SetName(shape->GetName());
-				bsOptShape->SetControllerRef(shape->GetControllerRef());
-				bsOptShape->SetSkinInstanceRef(shape->GetSkinInstanceRef());
-				bsOptShape->SetShaderPropertyRef(shape->GetShaderPropertyRef());
-				bsOptShape->SetAlphaPropertyRef(shape->GetAlphaPropertyRef());
-				bsOptShape->SetCollisionRef(shape->GetCollisionRef());
-				bsOptShape->GetProperties() = shape->GetProperties();
-				bsOptShape->GetExtraData() = shape->GetExtraData();
-
-				bsOptShape->transform = shape->transform;
-
-				bsOptShape->Create(vertices, &triangles, uvs, normals);
-				bsOptShape->flags = shape->flags;
+				//Only converting niTriShape to bsTriShape if required
+                if(options.bsTriShape || shape->HasType<NiTriStrips>() || shape->HasType<bhkMultiSphereShape>())
```

Comment:
What was the purpose for bhkMultiSphereShape being checked here?

Reply:
According to Zilav, bhkMultiSphereShape is not compatible with SSE.